pages/admin/AdminHandler.php

```diff
@@ -552,4 +568,154 @@ protected function getJobsTableState(int $page = 1): array
             'isLoadingItems' => false,
         ];
     }
+
+    /**
+     * List the queue jobs failied to execute
+     *
+     * @param array $args
+     * @param PKPRequest $request
+     */
+    public function failedJobs($args, $request)
+    {
+        $this->setupTemplate($request, true);
+
+        $templateMgr = TemplateManager::getManager($request);
+
+        $page = (int) ($request->getUserVar('page') ?? 1);
+
+        $breadcrumbs = $templateMgr->getTemplateVars('breadcrumbs');
+        $breadcrumbs[] = [
+            'id' => 'failedJobs',
+            'name' => __('navigation.tools.jobs.failed'),
+        ];
+
+        $templateMgr->setState($this->getFailedJobsTableState($request, $page));
+
+        $templateMgr->assign([
+            'pageComponent' => 'FailedJobsPage',
+            'breadcrumbs' => $breadcrumbs,
+            'pageTitle' => __('navigation.tools.jobs.failed'),
+        ]);
+
+        $templateMgr->display('admin/failedJobs.tpl');
+    }
+
+    /**
+     * Build the state data for the queued jobs table
+     */
+    protected function getFailedJobsTableState(PKPRequest $request, int $page = 1): array
+    {
+        $total = Repo::failedJob()->total();
+
+        $failedJobs = Repo::failedJob()
+            ->setOutputFormat(Repo::failedJob()::OUTPUT_HTTP)
+            ->perPage(self::JOBS_PER_PAGE)
+            ->setPage($page)
+            ->showJobs();
+
+        return [
+            'label' => __('navigation.tools.jobs.failed.view'),
+            'description' => __('admin.jobs.failed.totalCount', ['total' => $total]),
+            'columns' => [
+                [
+                    'name' => 'id',
+                    'label' => __('admin.jobs.list.id'),
+                    'value' => 'id',
+                ],
+                [
+                    'name' => 'title',
+                    'label' => __('admin.jobs.list.displayName'),
+                    'value' => 'displayName',
+                ],
+                [
+                    'name' => 'queue',
+                    'label' => __('admin.jobs.list.queue'),
+                    'value' => 'queue',
+                ],
+                [
+                    'name' => 'connection',
+                    'label' => __('admin.jobs.list.connection'),
+                    'value' => 'connection',
+                ],
+                [
+                    'name' => 'failed_at',
+                    'label' => __('admin.jobs.list.failedAt'),
+                    'value' => 'failed_at',
+                ],
+                [
+                    'name' => 'actions',
+                    'label' => __('admin.jobs.list.actions'),
+                    'value' => 'action',
+                ],
+            ],
+            'rows' => collect($failedJobs->all())->map(fn($failedJob) => array_merge($failedJob, [
```

Comment:
It would be nice if the list of jobs was fetched via the API as well, but I'll leave that aspect to Nate to make sure this proposal is consistent with other examples.

Reply:
Fixed.

Reply:
I also agree with this idea . wondering what is the best approach to make make a request form a page route/handler to api/handler route ?  in most of cases I can see that we make such api request from the view to API end point . But here we are passing the required base data into the initialisation of `VUE` component . There is always way such as we can fetch those in the component initialisation life cycle methods such as `created` or `mounted` but that feels bit unconventional approach to me . Any other idea or may be I am over thinking it .

Reply:
Yeah this is basically the way we do it. The fetch is extracted to a repository or some other method, then that's called in the APIHandler and the PageHandler. In this case, I don't think it's very important to pre-populate the data in the page, so you could load the data from the API in the Vue.js component once the page is loaded (in the `mounted()` callback).

Reply:
Done . All data fetch moved to API .